packages/backend-data/src/factory.test.ts

```diff
@@ -470,6 +470,25 @@ void describe('DataFactory', () => {
     );
   });
 
+  void it('throws on gen 1 patterns', () => {
+    resetFactoryCount();
+    const schema = `
+      type Profile @model {
+        author: Author @belongsTo
+      }
+      type Author @model {
+        profileID: ID
+        profile: Profile @hasOne(fields: ["profileID"])
+      }
+    `;
+    const dataFactory = defineData({
+      schema,
+    });
+    assert.throws(() => dataFactory.getInstance(getInstanceProps), {
+      message: 'Failed to instantiate data construct',
```

Comment:
Can we throw an error message that explains what is not supported and how to fix it?